pandas/core/indexes/range.py

```diff
@@ -215,6 +221,9 @@ def _format_data(self, name=None):
         # we are formatting thru the attributes
         return None
 
+    def _format_with_header(self, header, na_rep='NaN', **kwargs):
+        return header + [pprint_thing(x) for x in self._range]
```

Comment:
Without this I found that reprs of small DataFrames call RangeIndex.values and therefore RangeIndex._data. This avoids that.

Reply:
could do
```header + list(map(pprint_thing, self._range))```